config/read.go

```diff
@@ -420,6 +420,14 @@ func Read(logger *util.Logger, filename string) (Config, error) {
 					conf.Servers = append(conf.Servers, *config)
 				}
 			}
+
+			if config.DbURL != "" {
+				_, err := url.Parse(config.DbURL)
+				if err != nil {
+					prefixedLogger := logger.WithPrefix(config.SectionName)
+					prefixedLogger.PrintError("Could not parse db_url")
```

Comment:
Maybe clarify as "Could not parse db_url, check URL format and that special characters are correctly escaped"

Reply:
Well, the thing is, quoting and escaping values in go-ini are both pretty broken (see https://github.com/pganalyze/collector/issues/124#issuecomment-749835070 ), so I don't think that's going to be very helpful.

However, I went looking to see if I could fix this upstream, and while that's probably not worth it (there are a lot of different flags and behaviors that might break), I did find an option that works around the issue for the most part: with `SpaceBeforeInlineComment` set to true, a `#` only begins a single-line comment if it's preceded by a space. While theoretically this could break some configurations, I think it's fairly unlikely anyone was relying on this behavior (starting a line comment *without* offsetting it from the value with a space) in practice.

In the course of this, I realized that the real issue here is that the problem character needs to be percent-encoded, so most of this is moot. But I think we can still improve the message here along the lines of your suggestion.

I also was wondering if we should tell you what the error is, but unfortunately, Go can sometimes include part of the input string in the error message:
```
maciek@starling:~/duboce-labs/collector$ grep db_url t.conf
db_url             = postgres://foo:bar#hello@localhost/dbname
maciek@starling:~/duboce-labs/collector$ ./pganalyze-collector --config=t.conf --dry-run
2020/12/28 11:48:27 E [server1] Could not parse db_url: parse "postgres://foo:bar": invalid port ":bar" after host; note that any special characters must be percent-encoded
2020/12/28 11:48:27 E Config Error: Configuration file is empty, please edit t.conf and reload the collector
```

Since this output could easily end up in your logs, that's probably not a good idea. (Also, the message here suggests the URL is still being cut off, but that's not the case: perhaps Go is only printing the substring because it considers the rest of the URL a part of the fragment, which is not relevant to this particular parsing problem.)

Since the PR is approved, I'll merge with an improved error message (without the `url.Parse` error), but let me know if you have any further thoughts.

Reply:
Makes sense re: why we can't word this like suggested. Also agreed that we should be careful with the error message containing any sensitive data, so passing through the original error message is probably not a good idea. I think this is good to go 👍